optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/pages/rotation/RotationViewport.java

```diff
@@ -0,0 +1,137 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.openshift.employeerostering.gwtui.client.pages.rotation;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.jboss.errai.common.client.api.elemental2.IsElement;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.grid.CssGridLines;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.grid.Ticks;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Blob;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Lane;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.LinearScale;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Orientation;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.SubLane;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Viewport;
+import org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.view.BlobView;
+import org.optaplanner.openshift.employeerostering.shared.shift.Shift;
+import org.optaplanner.openshift.employeerostering.shared.spot.Spot;
+import org.optaplanner.openshift.employeerostering.shared.timeslot.TimeSlot;
+
+import static java.util.Collections.singletonList;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.rostergrid.model.Orientation.HORIZONTAL;
+
+public class RotationViewport extends Viewport<Long> {
+
+    private final Integer tenantId;
+    private final LocalDateTime baseDate;
+    private final Supplier<ShiftBlobView> blobViewFactory;
+    private final LinearScale<Long> scale;
+    private final CssGridLines gridLines;
+    private final Ticks<Long> ticks;
+    private final List<Lane<Long>> lanes;
+
+    RotationViewport(final Integer tenantId,
+                     final LocalDateTime baseDate,
+                     final Supplier<ShiftBlobView> blobViewFactory,
+                     final LinearScale<Long> scale,
+                     final CssGridLines gridLines,
+                     final Ticks<Long> ticks,
+                     final List<Lane<Long>> lanes) {
+
+        this.tenantId = tenantId;
+        this.baseDate = baseDate;
+        this.blobViewFactory = blobViewFactory;
+        this.scale = scale;
+        this.ticks = ticks;
+        this.lanes = lanes;
+        this.gridLines = gridLines;
+    }
+
+    @Override
+    public void drawGridLinesAt(final IsElement target) {
+        gridLines.drawAt(target, this);
+    }
+
+    @Override
+    public void drawTicksAt(final IsElement target) {
+        //FIXME: Make it18n
+        ticks.drawAt(target, this, minutes -> {
+            final Long hours = (minutes / 60) % 24;
```

Comment:
That's a code I didn't spend much time working on because it's very likely that it will change, so I think we should probably go for a `DataFormat` solution.

Just out of curiosity, the `minutes` variable represent the offset in minutes related to the baseDate. By making `(minutes % scale.getEndInScaleUnits())` I can get the day number having the last day to be 0, because it's a pattern.

Reply:
Okay, no need to clean this up if we're going to replace it soon.